### PR TITLE
feat: Add AtomaAPI chat completions JSON schema and HTTP server

### LINKED ISSUE #70: Add chat completions API based on OpenAPI
https://platform.openai.com/docs/api-reference/chat/create

### LINKED ISSUE #71: Create an HTTP web server for handling model requests
Requirements:
- use Axum
- route for handling POST requests for `/chat/completions`

### LINKED ISSUE #72: Create a JSON schema for validating model requests
Json schemas are great for validating some json that has required fields and structure, but also for documenting verbose APIs.

### PR DESCRIPTION
**Pull request by eureka-cpu**
Closes #70 
Closes #71 
Closes #72 

Adds an axum server that can handle two types of POST requests:
1. `/chat/completions/validate`: json validation against a schema for chat completions
2. `/chat/completions`: model chat completions

Adds the AtomaAPI, mostly ascribed from OpenAI's OpenAPI, which is derived from rust types as a json schema.

### CODE REVIEW

File: server/src/api/chat_completions.rs [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+// the `Role` is defining the type of `Message`. This may need some clarification either via
+// documentation to distinguish the differences between the OpenAPI and AtomaAPI, or further
+// work to make them the same.
+
+/// ID of the model to use.
+#[derive(Debug, Serialize, Deserialize, JsonSchema)]
+#[serde(rename(serialize = "model", deserialize = "model"))]
+pub enum Model {
+    #[serde(rename(serialize = "llama3", deserialize = "llama3"))]
+    Llama3,
```
Comment (jorgeantonio21) [inline server/src/api/chat_completions.rs:19]: We will include many different models, including different versions of the most recent Llama release (llama3.1 8b, 70b and 405b) and even previous version of Llama.

**Review by jorgeantonio21 — COMMENTED** [review]
This is looking great ! Left a few comments, I think some tiny adjustments are needed and it gets ready.

File: server/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+[package]
+name = "atoma-node-inference-server"
+version.workspace = true
+edition = "2021"
+
+[dependencies]
+anyhow.workspace = true
```
[the remark this answers is not in the document]
Comment (eureka-cpu) [reply server/Cargo.toml:7]: I've imported the ones that are shared from the workspace toml. IIRC It's more efficient for compile times if shared dependencies are in the workspace, but the inverse is true if you just add all dependencies to the workspace toml.

File: server/src/api/chat_completions.rs (hunk 1 above)
Comment (eureka-cpu) [reply server/src/api/chat_completions.rs:19]: Right, this is just a stub, I think I have a TODO somewhere referencing this.

**Comment by eureka-cpu** [comment]
@jorgeantonio21 if you wouldn't mind running the workflow I added to check that it's functioning I believe that's all that's left to do and this should be ready to go.

**Review by jorgeantonio21 — APPROVED** [review]
LGTM

**Pull request merged**